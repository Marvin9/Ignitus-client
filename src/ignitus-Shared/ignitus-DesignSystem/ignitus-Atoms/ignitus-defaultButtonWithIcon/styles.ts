import styled from '@emotion/styled';
import Icon from '../../../ignitus-Utilities/Components/icon';
import {
  ButtonProperties,
  getColor,
  getIconDimension,
} from '../../../ignitus-Utilities/HelperFunctions/emotionHelpers';

export const StyledIcon = styled(Icon)<ButtonProperties>`
  width: ${props => getIconDimension(props)};
  fill: ${props => getColor(props)};
`;
