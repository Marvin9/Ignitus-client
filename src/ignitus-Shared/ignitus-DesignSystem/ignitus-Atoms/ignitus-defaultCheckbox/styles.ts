import styled from '@emotion/styled';
import {Paragraph} from '../typography';
import {MD} from '../fonts';

export const CheckBox = styled.input`
  height: 1rem;
  width: 1rem;
  outline: none;
`;
export const Label = styled.label`
  margin-left: 0.4rem;
`;

export const StyledParagraph = styled(Paragraph)`
  font-size: ${MD};
`;
