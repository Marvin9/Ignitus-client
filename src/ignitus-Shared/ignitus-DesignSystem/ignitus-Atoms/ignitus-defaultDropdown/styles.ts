import styled from '@emotion/styled';
import {White, IgnitusBlue} from '../colors';
import {XS} from '../fonts';
import {DropDownProps} from './types';

export const Select = styled.select<DropDownProps>`
  display: ${props => props.display};
  margin-left: ${props => (props.display === 'inline-block' ? '1rem' : '0')};
  background-color: ${White};
  border: 2px solid ${IgnitusBlue};
  border-radius: 12px;
  font-size: ${XS};
  height: 2rem;
  width: 18rem;
  outline: none;
`;
