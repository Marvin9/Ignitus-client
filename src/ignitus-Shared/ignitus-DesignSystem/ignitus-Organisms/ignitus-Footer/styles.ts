import styled from '@emotion/styled';
import {maximumWidthQuery} from '../../ignitus-Atoms/media';
import {Heading3} from '../../ignitus-Atoms/typography';
import {White, SecondaryColor, IgnitusBlue} from '../../ignitus-Atoms/colors';
import {FooterLinkprops} from './types';
import {default as I} from '../../../ignitus-Utilities/Components/icon';

export const StyledHeading = styled(Heading3)`
  color: ${White};
  ${maximumWidthQuery[3]} {
    cursor: pointer;
  }
`;

export const Footer = styled.footer`
  background-color: ${IgnitusBlue};
  padding-bottom: 1rem;
  padding-top: 2rem;
`;

export const Container = styled.div`
  align-items: flex-start;
  display: flex;
  justify-content: space-evenly;
  margin: 0 auto;
  max-width: 72rem;
  min-height: 20vh;
  ${maximumWidthQuery[3]} {
    align-items: center;
    flex-direction: column;
    justify-content: center;
  }
`;

export const FooterSections = styled.div`
  margin: 0 2rem;
  width: max-content;
  h5 {
    color: ${White};
    font-weight: bold;
  }
  ${maximumWidthQuery[3]} {
    align-self: center;
    padding: 1rem;
    margin: 0;
    margin-bottom: 0.5rem;
    max-width: 15rem;
    text-align: center;
    transition: max-height 1s ease-in;
  }
`;

export const Logo = styled(I)`
  margin-top: 1rem;
  width: 5rem;
  ${maximumWidthQuery[3]} {
    margin: 1rem;
  }
  path {
    fill: ${White};
  }
`;

export const FooterLinks = styled('ul')<FooterLinkprops>`
  display: flex;
  flex-direction: column;
  li {
    padding: 2px;
    ${maximumWidthQuery[3]} {
      align-self: center;
    }
    a {
      &:link,
      &:visited {
        color: ${White};
      }
    }
    a {
      &:hover,
      &:active {
        color: ${SecondaryColor};
      }
    }
  }
  ${maximumWidthQuery[3]} {
    display: ${props => (props.toggle ? 'none' : 'block')};
  }
`;

export const Socials = styled.div`
  margin: 1rem auto;
  max-width: max-content;
`;

export const FooterSocial = styled.ul`
  display: flex;
  li {
    padding: 0rem 0.5rem;
    a {
      &:link,
      &:visited {
        color: ${White};
      }
    }
    a {
      &:hover,
      &:active {
        color: ${SecondaryColor};
      }
    }
  }
`;

export const Icon = styled(I)`
  width: 1rem;
  height: 1rem;
  fill: ${White};
  &:hover {
    fill: ${SecondaryColor};
  }
`;

export const Copyright = styled.p`
  color: ${White};
  font-weight: bold;
  letter-spacing: 1px;
  text-align: center;
`;
