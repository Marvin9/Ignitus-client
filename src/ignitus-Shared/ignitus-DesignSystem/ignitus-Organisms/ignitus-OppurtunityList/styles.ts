import styled from '@emotion/styled';
import * as T from '../../ignitus-Atoms/typography';
import * as C from '../../ignitus-Atoms/colors';
import {flexibleColDiv, flexibleRowDiv, Paragraph} from '../../shared';
import {Normal, MD} from '../../ignitus-Atoms/fonts';
import ApplyIcon from '../../ignitus-Assets/ignitus-Icons/applyIcon/applyIcon';

export const OpportunityList = styled(flexibleColDiv)``;

export const OpportunityContainer = styled(flexibleRowDiv)`
  border-radius: 4px;
  box-shadow: 0 2px 4px 0 ${C.boxShadowColor};
  background-color: ${C.White};
  padding: 1rem;
  width: 100%;
`;

export const DateBtnContainer = styled(flexibleColDiv)`
  padding: 0.5rem;
  border: 2px solid blue;
  flex: 1;
`;

export const DateContainer = styled(T.Heading6)`
  color: ${C.GreyOne};
  font-weight: ${Normal};
`;

export const BtnContainer = styled(flexibleRowDiv)`
  padding: 5px 10px;
  border: 2px solid ${C.IgnitusBlue};
  border-radius: 5px;
  cursor: pointer;
`;

export const BtnContainerText = styled(Paragraph)`
  font-size: ${MD};
  font-weight: ${Normal};
  margin-bottom: 0;
  margin-left: 10px;
  color: ${C.IgnitusBlue};
`;

export const BtnContainerIcon = styled(ApplyIcon)`
  height: 1.5rem;
  fill: ${C.IgnitusBlue};
`;

export const ImageDescriptionContainer = styled(flexibleRowDiv)`
  flex: 6;
`;

export const ImageContainer = styled(flexibleRowDiv)`
  flex: 1;
  img {
    max-width: 100%;
    max-height: 100%;
  }
`;

export const DescriptionContainer = styled.div`
  display: flex;
  flex-direction: column;
  flex: 5;
  padding-left: 1rem;
`;

export const OpportunityDescriptionBtnContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

export const OpportunityDescription = styled(T.Paragraph)`
  font-weight: ${Normal};
  color: ${C.GreyText};
  margin-top: 1rem;
`;

export const OpportunityDateTitle = styled.div`
  display: flex;
  sflex-direction: row;
`;

export const OpportunityTitle = styled(T.Heading3)`
  flex: 1;
`;

export const OpportunityData = styled(T.Paragraph)`
  color: ${C.GreyText};
  font-weight: ${Normal};
  flex: 1;
  text-align: end;
`;

export const OpportunitySubTitle = styled(T.Paragraph)`
  font-weight: ${Normal};
  color: ${C.GreyOne};
`;
