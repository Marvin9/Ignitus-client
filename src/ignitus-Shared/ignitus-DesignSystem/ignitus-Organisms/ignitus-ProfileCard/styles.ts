import styled from '@emotion/styled';
import * as C from '../../ignitus-Atoms/colors';

import {
  flexibleColDiv,
  flexibleRowDiv,
  Link as L,
  Paragraph as P,
} from '../../shared';
import {Heading1, Heading4} from '../../ignitus-Atoms/typography';
import {XXXL, SemiBold, Normal} from '../../ignitus-Atoms/fonts';

import {default as I} from '../../../ignitus-Utilities/Components/icon';
import {ParagraphProps} from './types';

export const Container = styled.div`
  display: flex;
  width: 60rem;
  flex-direction: column;
  padding: 1rem 2rem;
  border-radius: 1rem;
  box-shadow: 0 2px 4px 0 ${C.boxShadowColor};
  background-color: ${C.White};
  color: ${C.IgnitusBlue};
`;

export const TopSection = styled.div`
  text-align: end;
`;

export const BottomSection = styled.div`
  display: flex;
  flex-direction: row;
  padding: 1rem;
`;

export const LeftRow = styled.div`
  display: flex;
  flex-direction: row;
  flex: 2;
  justify-content: flex-start;
`;

export const Avatar = styled.img`
  width: 10rem;
  border-radius: 1rem;
`;

export const MiddleRow = styled.div`
  display: flex;
  flex-direction: column;
`;

export const ImageContainer = styled(flexibleRowDiv)`
  width: 200px;
  height: 190px;
  img {
    width: 100%;
    height: 100%;
    border-radius: 15px;
  }
`;

export const Paragraph = styled(P)<ParagraphProps>`
  color: ${(props) => (props.primary ? C.IgnitusBlue : C.GreySecondaryText)};
  font-weight: ${Normal};
  display: inline;
  margin: ${(props) => props.margin};
`;

export const Name = styled(Heading1)`
  font-size: ${XXXL};
  font-weight: ${SemiBold};
`;

export const Designation = styled(Heading4)`
  margin-bottom: 1rem;
  margin-top: 1rem;
`;

export const ProfileInformationContainer = styled.div`
  margin-left: 4rem;
`;

export const RightRow = styled(flexibleColDiv)`
  flex: 1;
  align-items: flex-start;
`;

export const Contact = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

export const Icons = styled(flexibleRowDiv)`
  margin-bottom: 2rem;
`;

export const Link = styled(L)`
  margin-right: 2rem;
  display: inline-block;
`;

export const Icon = styled(I)`
  width: 1.5rem;
  fill: ${C.IgnitusBlue};
  path {
  }
`;

export const Wrapper = styled.div`
  display: flex;
  margin: 0.6rem 0;
  flex-direction: row;
`;

export const DropDownBtnContainer = styled.div`
  position: relative;
  margin-top: 1rem;
`;

export const DropDownBtn = styled.div`
  font-weight: bold;
  color: ${C.GreyOne};
  cursor: pointer;
  display: inline-block;
  padding-bottom: 10px;
  transition: all 0.4s;
  width: fit-content;

  &:hover {
    ul {
      display: block;
    }
  }
`;

export const DropDownMenu = styled.ul`
  z-index: 3;
  position: absolute;
  top: 30px;
  left: 0;
  border-radius: 10px 0 10px 10px;
  box-shadow: 0 2px 4px 0 ${C.boxShadowColor};
  background-color: ${C.White};
  color: ${C.IgnitusBlue};
  display: none;
  font-weight: 400;

  &:hover {
    display: block;
  }
`;

export const DropDownMenuListItem = styled.li`
  padding: 0.5em 1em;
  font-size: 14px;
  width: 200px;

  &:first-of-type {
    border-top-left-radius: 10px;
  }

  &:last-child {
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
  }

  &:hover {
    color: ${C.White};
    background-color: ${C.IgnitusBlue};
    cursor: pointer;
  }
`;
