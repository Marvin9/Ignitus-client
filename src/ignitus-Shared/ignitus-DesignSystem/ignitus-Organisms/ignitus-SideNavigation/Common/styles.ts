import styled from '@emotion/styled';

import { White, IgnitusBlue, GreyLight } from '../../../ignitus-Atoms/colors';
import { MD, Normal } from '../../../ignitus-Atoms/fonts';
import { StyledLayerProps } from './types';

export const Layer = styled.div<StyledLayerProps>`
  border-bottom: ${props => (props.border ? `1px solid ${GreyLight}` : 'null')};
  color: ${IgnitusBlue};
  cursor: pointer;
  font-size: ${MD};
  font-weight: ${Normal};
  margin-top: ${props => (props.marginTop ? '0.5rem' : 'none')};
  margin-bottom: ${props => (props.marginBottom ? '0.5rem' : 'none')};
  padding: 1rem 0;
  width: 100%;
  &:hover {
    background-color: ${IgnitusBlue};
    p {
      color: ${White};
    }
    ${props =>
      props.containIcon
        ? `
        svg {
          fill: ${White};
        }`
        : ''}
  }
`;
