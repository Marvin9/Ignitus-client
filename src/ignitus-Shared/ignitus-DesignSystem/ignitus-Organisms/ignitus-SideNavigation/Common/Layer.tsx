import React from 'react';

import { Paragraph, IgnitusBlue } from '../../../../index';
import * as S from './styles';
import { LayerProps } from './types';

export const CommonLayer: React.FC<LayerProps> = ({
  text,
  border = true,
  marginTop = false,
  marginBottom = false,
  append,
  prepend,
  ...rest
}) => (
  <S.Layer
    border={border}
    marginTop={marginTop}
    marginBottom={marginBottom}
    containIcon={!!prepend}
    {...rest}
  >
    {!!prepend && prepend}
    <Paragraph color={IgnitusBlue}>{text}</Paragraph>
    {!!append && append}
  </S.Layer>
);
