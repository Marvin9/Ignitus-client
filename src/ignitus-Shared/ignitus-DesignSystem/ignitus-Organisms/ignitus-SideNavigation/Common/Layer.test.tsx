import React from 'react';
import { mount, render } from 'enzyme';
import styled from '@emotion/styled';
import serializer, { matchers } from 'jest-emotion';

import { GreyLight } from '../../../../index';

import { CommonLayer } from './Layer';
import { Layer } from './styles';

expect.addSnapshotSerializer(serializer);
expect.extend(matchers);

let wrapper;

describe('<Layer />', () => {
  it('should render', () => {
    wrapper = render(<CommonLayer text="Text" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with default border-bottom style', () => {
    wrapper = mount(<CommonLayer text="Text" />);
    expect(wrapper.find(Layer)).toHaveStyleRule(
      'border-bottom',
      `1px solid ${GreyLight}`,
    );
  });

  it('should render correct styles given in props', () => {
    wrapper = mount(<CommonLayer text="Text" marginTop marginBottom />);
    const margin = '0.5rem';
    expect(wrapper).toHaveStyleRule('margin-top', margin);
    expect(wrapper).toHaveStyleRule('margin-bottom', margin);
  });

  it('should update style using emotion styled', () => {
    const StyledLayer = styled(CommonLayer)`
      background-color: red;
    `;
    wrapper = mount(<StyledLayer text="Text" />);
    expect(wrapper).toHaveStyleRule('background-color', 'red');
  });

  it('should render text in Paragraph', () => {
    const text = 'Text';
    wrapper = mount(<CommonLayer text={text} />);
    expect(wrapper.find('p').text()).toBe(text);
  });

  it('should render appended & prepended React Node passed in props', () => {
    const append = <h1>Appended Heading</h1>;
    const prepend = <h1>Prepended Heading</h1>;
    wrapper = mount(
      <CommonLayer text="Hello" append={append} prepend={prepend} />,
    );

    // .find(Layer).children() => <div>
    // .find(Layer).children().children() => Children of Layer
    const LayerChild = wrapper
      .find(Layer)
      .children()
      .children();
    expect(LayerChild.length).toBe(3); // Including Text paragraph

    expect(LayerChild.at(0).equals(prepend)).toBeTruthy();
    expect(LayerChild.at(2).equals(append)).toBeTruthy();
  });
});
