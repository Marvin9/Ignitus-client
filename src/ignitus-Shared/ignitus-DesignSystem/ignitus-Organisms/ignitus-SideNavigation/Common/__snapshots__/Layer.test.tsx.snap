// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layer /> should render 1`] = `
.emotion-1 {
  border-bottom: 1px solid #E0E0E0;
  color: #000066;
  cursor: pointer;
  font-size: 18px;
  font-weight: 400;
  margin-top: none;
  margin-bottom: none;
  padding: 1rem 0;
  width: 100%;
}

.emotion-1:hover {
  background-color: #000066;
}

.emotion-1:hover p {
  color: #ffffff;
}

.emotion-0 {
  font-size: 16px;
  font-weight: 400;
  color: #000066;
  margin: 0;
}

<div
  class="emotion-1"
>
  <p
    class="emotion-0"
    color="#000066"
  >
    Text
  </p>
</div>
`;
